### ISSUE TITLE
✅ 🐞 Autoplay on video lightbox doesn't work.

### ISSUE THREAD

**Issue by Nazrinn** [elementor/elementor#14437]
## Prerequisites

- [x] I have searched for similar issues in both open and closed tickets and cannot find a duplicate.
- [x] The issue still exists against the latest stable version of Elementor.

## Description
YouTube autoplay doesn't work on video lightbox (at least).

## Steps to reproduce
Create any video lightbox and set the autoplay parameter to 1. Autoplay will only work *sometimes* on first lightbox trigger. Autoplay seem to always work on second video lightbox trigger, regardless of the video lightbox being triggered.

## Additional details (potential fix!)
Since 2018, Google changed its autoplay policies in a way that using the YouTube parameter `autoplay=1` isn't sufficient to trigger autoplay on videos. You have to add `allow="autoplay"` attribute to the video iframe in order to authorize the video to autoplay without user interaction, (or mute the video, although the former is probably preferable for video lightbox). 

Looking forward seeing that being fixed in future versions.

[Source of fix (Stack Overflow)](https://stackoverflow.com/questions/40685142/youtube-autoplay-not-working)
[Google documentation on autoplay policy](https://developers.google.com/web/updates/2017/09/autoplay-policy-changes)

## Isolating the problem
- [x] This bug happens with only Elementor plugin active (and Elementor Pro).
- [x] This bug happens with a default WordPress theme active.
- [x] I can reproduce this bug consistently using the steps above.

## Environment

<details>
<summary>Click to reveal</summary>

````txt

== Server Environment ==
	Operating System: WINNT
	Software: Apache/2.4.46 (Win64) OpenSSL/1.1.1j PHP/8.0.3
	MySQL version: mariadb.org binary distribution v10.4.18
	PHP Version: 8.0.3
	PHP Max Input Vars: 1000
	PHP Max Post Size: 128M
	GD Installed: Yes
	ZIP Installed: Yes
	Write Permissions: All right
	Elementor Library: Connected

== WordPress Environment ==
	Version: 5.7
	Site URL: http://localhost/dev/barbandcarole
	Home URL: http://localhost/dev/barbandcarole
	WP Multisite: No
	Max Upload Size: 128 MB
	Memory limit: 512M
	Permalink Structure: /%year%/%monthnum%/%day%/%postname%/
	Language: en-CA
	Timezone: America/Toronto
	Debug Mode: Active

== Theme ==
	Name: Hello Elementor Child
	Version: 1.0.1
	Author: Elementor Team
	Child Theme: Yes
	Parent Theme Name: Hello Elementor
	Parent Theme Version: 2.3.1
	Parent Theme Author: Elementor Team

== User ==
	Role: administrator
	WP Profile lang: en_CA
	User Agent: Mozilla/5.0 (Windows NT 10.0; Win64; x64) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/89.0.4389.114 Safari/537.36

== Active Plugins ==
	Advanced Custom Fields PRO
		Version: 5.9.5
		Author: Elliot Condon

	Custom Post Type UI
		Version: 1.9.0
		Author: WebDevStudios

	Elementor
		Version: 3.1.4
		Author: Elementor.com

	Elementor Pro
		Version: 3.3.0-dev1
		Author: Elementor.com

	Enable Media Replace
		Version: 3.5.0
		Author: ShortPixel

	Phoenix Media Rename
		Version: 3.2.0
		Author: crossi72

	Post Views Counter
		Version: 1.3.4
		Author: Digital Factory

	Simple Custom Post Order
		Version: 2.5.4
		Author: Colorlib

	SVG Support
		Version: 2.3.18
		Author: Benbodhi

	WPvivid Backup Plugin
		Version: 0.9.52
		Author: WPvivid Team

	Yoast Duplicate Post
		Version: 4.1.2
		Author: Enrico Battocchi & Team Yoast


== Elementor - Compatibility Tag ==
	
	Elementor Pro: compatible

== Elementor Pro - Compatibility Tag ==
	


== Elements Usage ==
	
	archive : 1
		button : 2
		heading : 2
		html : 3
		image : 1
		shortcode : 1
		text-editor : 2
	footer : 1
		form : 1
		heading : 7
		image : 4
		shortcode : 4
		social-icons : 1
		spacer : 1
		text-editor : 6
	header : 1
		nav-menu : 1
		social-icons : 1
		theme-site-logo : 1
	section : 3
		button : 2
		heading : 2
		html : 3
		image : 1
		text-editor : 2
	widget : 0
		html : 69
		shortcode : 55
	wp-page : 5
		button : 23
		counter : 2
		divider : 2
		form : 1
		global : 1
		heading : 44
		html : 60
		image : 16
		media-carousel : 1
		shortcode : 14
		social-icons : 2
		spacer : 17
		text-editor : 32
	single-post : 1
		button : 2
		heading : 6
		html : 5
		image-carousel : 1
		shortcode : 12
		text-editor : 1
		video : 1


== Log ==
	
Log: showing 15 of 152021-03-01 02:29:18 [info] Elementor data updater process has been completed. [array (
  'plugin' => 'Elementor',
  'from' => '3.1.1',
  'to' => '3.2.0-dev4',
)]
2021-03-02 19:05:32 [info] Elementor data updater process has been completed. [array (
  'plugin' => 'Elementor',
  'from' => '3.2.0-dev4',
  'to' => '3.2.0-dev5',
)]
2021-03-02 19:14:01 [info] Elementor data updater process has been completed. [array (
  'plugin' => 'Elementor Pro',
  'from' => '3.1.1',
  'to' => '3.2.0-dev5',
)]
2021-03-04 00:14:26 [info] Elementor data updater process has been completed. [array (
  'plugin' => 'Elementor',
  'from' => '3.2.0-dev5',
  'to' => '3.2.0-dev6',
)]
2021-03-08 19:27:57 [info] Elementor data updater process has been completed. [array (
  'plugin' => 'Elementor',
  'from' => '3.2.0-dev6',
  'to' => '3.2.0-dev7',
)]
2021-03-08 20:05:37 [info] Elementor data updater process has been completed. [array (
  'plugin' => 'Elementor Pro',
  'from' => '3.2.0-dev5',
  'to' => '3.2.0-dev6',
)]
2021-03-10 22:44:25 [info] Elementor data updater process has been completed. [array (
  'plugin' => 'Elementor',
  'from' => '3.2.0-dev7',
  'to' => '3.2.0-dev8',
)]
2021-03-16 12:27:03 [info] Elementor data updater process has been completed. [array (
  'plugin' => 'Elementor Pro',
  'from' => '3.2.0-dev6',
  'to' => '3.2.0-dev7',
)]
2021-03-16 13:53:14 [info] Elementor data updater process has been completed. [array (
  'plugin' => 'Elementor Pro',
  'from' => '3.2.0-dev7',
  'to' => '3.2.0',
)]
2021-03-17 02:35:23 [info] Elementor data updater process has been completed. [array (
  'plugin' => 'Elementor Pro',
  'from' => '3.2.0',
  'to' => '3.3.0-dev1',
)]
2021-03-22 14:35:34 [info] elementor::elementor_updater Started 
2021-03-22 14:35:34 [info] Elementor/Upgrades - _v_3_2_0_migrate_breakpoints_to_new_system Start  
2021-03-22 14:35:34 [info] Elementor/Upgrades - _v_3_2_0_migrate_breakpoints_to_new_system Finished 
2021-03-22 14:35:34 [info] Elementor data updater process has been completed. [array (
  'plugin' => 'Elementor',
  'from' => '3.2.0-dev8',
  'to' => '3.2.0-beta1',
)]
2021-03-22 14:35:34 [info] Elementor data updater process has been queued. [array (
  'plugin' => 'Elementor',
  'from' => '3.2.0-dev8',
  'to' => '3.2.0-beta1',
)]

PHP: showing 12 of 12PHP: 2021-03-01 02:38:33 [notice X 10][..\wp-content\plugins\elementor\core\common\modules\ajax\module.php::171] Undefined index: data [array (
  'trace' => '
#0: Elementor\Core\Logger\Manager -> shutdown()
',
)]
PHP: 2021-03-01 02:40:13 [notice X 1][..\wp-content\plugins\elementor\includes\managers\image.php::86] Trying to access array offset on value of type bool [array (
  'trace' => '
#0: Elementor\Core\Logger\Manager -> shutdown()
',
)]
PHP: 2021-03-01 03:16:16 [error X 4][..\wp-content\plugins\elementor-pro\modules\theme-builder\classes\conditions-manager.php::281] Uncaught Error: Call to a member function update_meta() on null in ..\wp-content\plugins\elementor-pro\modules\theme-builder\classes\conditions-manager.php:281
Stack trace:
#0 ..\wp-content\plugins\elementor-pro\modules\theme-builder\classes\conditions-manager.php(162): ElementorPro\Modules\ThemeBuilder\Classes\Conditions_Manager->save_conditions(7, Array)
#1 [internal function]: ElementorPro\Modules\ThemeBuilder\Classes\Conditions_Manager->ajax_save_theme_template_conditions(Array, Object(Elementor\Core\Common\Modules\Ajax\Module))
#2 ..\wp-content\plugins\elementor\core\common\modules\ajax\module.php(171): call_user_func(Array, Array, Object(Elementor\Core\Common\Modules\Ajax\Module))
#3 ..\wp-includes\class-wp-hook.php(287): Elementor\Core\Common\Modules\Ajax\Module->handle_ajax_request('')
#4 ..\wp-includes\class-wp-hook.php(311): WP_Hook->apply_filters [array (
  'trace' => '
#0: Elementor\Core\Logger\Manager -> shutdown()
',
)]
PHP: 2021-03-08 07:16:33 [notice X 38][..\wp-content\plugins\elementor-pro\modules\dynamic-tags\acf\tags\acf-text.php::33] Undefined offset: 1 [array (
  'trace' => '
#0: Elementor\Core\Logger\Manager -> shutdown()
',
)]
PHP: 2021-03-12 20:17:04 [warning X 1][..\wp-content\plugins\elementor-pro\modules\assets-manager\asset-types\icons\custom-icons.php::180] unlink(briefcase.svg): No such file or directory [array (
  'trace' => '
#0: Elementor\Core\Logger\Manager -> shutdown()
',
)]
PHP: 2021-03-15 04:12:19 [error X 2][..\wp-content\plugins\elementor-pro\modules\theme-builder\classes\conditions-manager.php::281] Uncaught Error: Call to a member function update_meta() on bool in ..\wp-content\plugins\elementor-pro\modules\theme-builder\classes\conditions-manager.php:281
Stack trace:
#0 ..\wp-content\plugins\elementor-pro\modules\theme-builder\classes\conditions-manager.php(162): ElementorPro\Modules\ThemeBuilder\Classes\Conditions_Manager->save_conditions(NULL, Array)
#1 [internal function]: ElementorPro\Modules\ThemeBuilder\Classes\Conditions_Manager->ajax_save_theme_template_conditions(Array, Object(Elementor\Core\Common\Modules\Ajax\Module))
#2 ..\wp-content\plugins\elementor\core\common\modules\ajax\module.php(171): call_user_func(Array, Array, Object(Elementor\Core\Common\Modules\Ajax\Module))
#3 ..\wp-includes\class-wp-hook.php(292): Elementor\Core\Common\Modules\Ajax\Module->handle_ajax_request('')
#4 ..\wp-includes\class-wp-hook.php(316): WP_Hook->apply_filt [array (
  'trace' => '
#0: Elementor\Core\Logger\Manager -> shutdown()
',
)]
PHP: 2021-03-17 14:32:45 [error X 2][..\wp-content\plugins\elementor-pro\modules\theme-builder\classes\conditions-manager.php::281] Uncaught Error: Call to a member function update_meta() on null in ..\wp-content\plugins\elementor-pro\modules\theme-builder\classes\conditions-manager.php:281
Stack trace:
#0 ..\wp-content\plugins\elementor-pro\modules\theme-builder\classes\conditions-manager.php(162): ElementorPro\Modules\ThemeBuilder\Classes\Conditions_Manager->save_conditions(7, Array)
#1 [internal function]: ElementorPro\Modules\ThemeBuilder\Classes\Conditions_Manager->ajax_save_theme_template_conditions(Array, Object(Elementor\Core\Common\Modules\Ajax\Module))
#2 ..\wp-content\plugins\elementor\core\common\modules\ajax\module.php(171): call_user_func(Array, Array, Object(Elementor\Core\Common\Modules\Ajax\Module))
#3 ..\wp-includes\class-wp-hook.php(292): Elementor\Core\Common\Modules\Ajax\Module->handle_ajax_request('')
#4 ..\wp-includes\class-wp-hook.php(316): WP_Hook->apply_filters [array (
  'trace' => '
#0: Elementor\Core\Logger\Manager -> shutdown()
',
)]
PHP: 2021-03-21 21:18:14 [notice X 162][..\wp-content\plugins\elementor\includes\conditions.php::90] Trying to access array offset on value of type null [array (
  'trace' => '
#0: Elementor\Core\Logger\Manager -> shutdown()
',
)]
PHP: 2021-03-30 21:36:25 [warning X 23][..\wp-content\plugins\elementor\includes\conditions.php::90] Trying to access array offset on value of type null [array (
  'trace' => '
#0: Elementor\Core\Logger\Manager -> shutdown()
',
)]
PHP: 2021-04-01 18:09:41 [warning X 1][..\wp-content\plugins\elementor-pro\modules\dynamic-tags\acf\tags\acf-url.php::34] Undefined array key 1 [array (
  'trace' => '
#0: Elementor\Core\Logger\Manager -> shutdown()
',
)]
PHP: 2021-04-03 23:29:38 [warning X 1][..\wp-content\plugins\elementor-pro\modules\assets-manager\asset-types\icons\custom-icons.php::243] unlink(../wp-content/uploads/2021/04/SVG.zip): Resource temporarily unavailable [array (
  'trace' => '
#0: Elementor\Core\Logger\Manager -> shutdown()
',
)]
PHP: 2021-04-03 23:30:50 [warning X 1][..\wp-content\plugins\elementor-pro\modules\assets-manager\asset-types\icons\custom-icons.php::243] unlink(../wp-content/uploads/2021/04/Fontello.zip): Resource temporarily unavailable [array (
  'trace' => '
#0: Elementor\Core\Logger\Manager -> shutdown()
',
)]

````

</details>

**Comment by shilo-ey** [comment]
Thanks!

We will take a look and update this thread.

**Comment by tissi-2** [comment]
+1

**Comment by rtpHarry** [comment]
It seems this never moved forward.

Looking at the code that is included when I add a lightbox dynamic tag, set to a video, with the autoplay=1, there is no allow attribute:

<img width="704" alt="image" src="https://user-images.githubusercontent.com/1038062/235697613-2f656de0-94d6-405d-b982-82c78234b0b0.png">

Is this markup being provided by Elementor Pro or is it something being done by WordPress Oembed (What I'm wondering is if it can be fixed by some kind of WordPress filter rather than waiting for Elementor)

**Comment by JxxIT** [comment]
@rtpHarry @tissi-2 @shilo-ey @Nazrinn My pull request is merged, #24324 this sould be fixed by the next update.

**Comment by nicholaszein** [comment]
Hello there!

We have great news! 🙌 

### 📢 We're happy to announce that the issue you raised was resolved in Elementor v3.23.0! 🥳 

✅ Feel free to check it out and update your plugin to the new version!

> _Check the changelog for more information:_
> [Changelog of Elementor and Elementor Pro](https://elementor.com/help/changelog-of-elementor/)

Cheers 🥂
